### PR TITLE
QGIS Raster analysis algs, table format for parameters, 3.4

### PR DESCRIPTION
**Pull request by havatv**
Goal:

Ticket(s): #4429

- [ ] Backport to LTR documentation is required

### CODE REVIEW

File: source/docs/user_manual/processing_algs/qgis/rasteranalysis.rst [hunk 1]
```diff
@@ -359,68 +666,154 @@ of the raster layer that intersects the polygon(s).
...
+     - ``INPUT_VECTOR``
+     - [vector: polygon]
+     - Pologyn vector layer that defines the zones.
+   * - **Output column prefix**
+       
+       Optional
+     - ``COLUMN_PREFIX``
+     - [string]
+       
+       Default: 'HISTO_'
```
Comment (havatv) [inline source/docs/user_manual/processing_algs/qgis/rasteranalysis.rst]:
```suggestion
       Default: 'HISTO\_'
```

**Comment by havatv** [comment]
@DelazJ, sorry for not making it clear that this is work in progress. Don't waste time on it until I have completed the whole document.

**Comment by DelazJ** [comment]
It has been removed yesterday so I thought it was OK. No problem, anyway.

File: source/docs/user_manual/processing_algs/qgis/rasteranalysis.rst [hunk 2]
```diff
@@ -167,74 +351,117 @@ specified in a vector table.
...
+     - Values that do not belong to a class will result in the
+       no data value.
+       If False, the original value is kept.
+   * - **Output data type**
+     - ``DATA_TYPE``
+     - [enumeration]
+       
+       Default: 5
+     - Defines the data type of the output raster file.
+       Options:
```
[the remark this answers is not in the document]
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/rasteranalysis.rst]: The problem is that the enumerations may be different even if the available data types are the same...
In this case, there are 10 entries, but for other algorithms (for instance the GDAL "Rearrange bands" algorithm) there are 11 entries (and the first one, 0, is "Use Input Layer Datatype").

File: source/docs/user_manual/processing_algs/qgis/rasteranalysis.rst [hunk 3]
```diff
@@ -247,66 +474,106 @@ specified in a fixed table.
...
```
[the remark this answers is not in the document]
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/rasteranalysis.rst]: See answer above.

File: source/docs/user_manual/processing_algs/qgis/rasteranalysis.rst [hunk 4]
```diff
@@ -95,29 +134,95 @@ The output is loaded in the :menuselection:`Processing --> Results viewer` menu.
...
+     - 
+   * - **Mean value**
+     - ``MEAN``
+     - [number]
+     - 
+   * - **Minimum value**
+     - ``MIN``
+     - [number]
+     - 
+   * - **Output**
```
[the remark this answers is not in the document]
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/rasteranalysis.rst]: The output comes sorted in the returned dictionary.  What we could do is to mark the output elements that will available in the results viewer and those that will be added to the map/TOC. I think that should be discussed in a new issue.

File: source/docs/user_manual/processing_algs/qgis/rasteranalysis.rst [hunk 5]
```diff
@@ -129,32 +234,111 @@ Returns the count and area of each unique value in a given raster layer.
...
+     - 
+   * - **NODATA pixel count**
+     - ``NODATA_PIXEL_COUNT``
+     - [number]
+     - 
+   * - **Total pixel count**
+     - ``TOTAL_PIXEL_COUNT``
+     - [number]
+     - 
+   * - **Unique values report**
```
[the remark this answers is not in the document]
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/rasteranalysis.rst]: The output comes sorted in the returned dictionary.  What we could do is to mark the output elements that will available in the results viewer and those that will be added to the map/TOC. I think that should be discussed in a new issue.

File: source/docs/user_manual/processing_algs/qgis/rasteranalysis.rst [hunk 6]
```diff
@@ -129,32 +234,111 @@ Returns the count and area of each unique value in a given raster layer.
...
+     - The output HTML file contains the following information:
+       
+       * Analyzed file: the path of the raster layer
+       * Extent: xmin, ymin, xmax, ymax coordinates of the extent
+       * Projection: projection of the layer
+       * Width in pixels: number of columns and pixel width size
+       * Height in pixels: number of rows and pixel width size
+       * Total pixel count: count of all the pixels
+       * NODATA pixel count: count of pixels with NODATA value
+   * - **Unique values table**
```
[the remark this answers is not in the document]
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/rasteranalysis.rst]: The output comes sorted in the returned dictionary.  What we could do is to mark the output elements that will available in the results viewer and those that will be added to the map/TOC. I think that should be discussed in a new issue.

**Comment by havatv** [comment]
I have too many non-merged PRs to worry about, so I must have mixed this with another one.  The PR was, in fact, complete (I was apparently checking the wrong code)...

File: source/docs/user_manual/processing_algs/qgis/rasteranalysis.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (DelazJ) [reply source/docs/user_manual/processing_algs/qgis/rasteranalysis.rst]: I know there are different lists (actually they are two), so why not create two lists (as far as I remember my previous investigation I could not find a way to derive one from the other)?

File: source/docs/user_manual/processing_algs/qgis/rasteranalysis.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (DelazJ) [reply source/docs/user_manual/processing_algs/qgis/rasteranalysis.rst]: I know you were following the output logic. Adding another visual and easy-to-guess element to distinguish items in a (somehow constrained) table structure would not be easy and lead to more discussion. And I'm not sure that readers would really care about the sorting here, but providing the most beneficial information first and not somewhere in the middle of the list might do.

File: source/docs/user_manual/processing_algs/qgis/rasteranalysis.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/rasteranalysis.rst]: We can, but we have not done so yet.  So until we have done that, I think we need to list the enumerations.

**Pull request merged**